tv_apps/smart-home/js/card_util.js

```diff
@@ -187,19 +188,16 @@
       var nameSpan = document.createElement('span');
       nameSpan.classList.add('name');
       this._localizeCardName(nameSpan, card);
-      cardButton.appendChild(nameSpan);
 
-      return cardButton;
+      var fragment = document.createDocumentFragment();
+      fragment.appendChild(cardButton);
+      fragment.appendChild(nameSpan);
+
+      return fragment;
     },
 
-    updateCardName: function(cardButton, card) {
-      var spans =
-           SharedUtils.nodeListToArray(cardButton.getElementsByTagName('span'));
-      spans.forEach(span => {
-        if (span.classList.contains('name')) {
-          this._localizeCardName(span, card);
-        }
-      });
+    updateCardName: function(nameElement, card) {
```

Comment:
According to the spec, we may need to update the second name span in the folder card, so I suggest you pass the entire card element as the first argument.